pkg/stub/handler.go

```diff
@@ -366,46 +476,60 @@ func (h *Handler) CreateDefaultResourceIfNeeded() error {
 			return false, nil
 		})
 		if err != nil {
-			return h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "issues waiting for delete to complete: %v")
+			return nil, h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionUnknown, err, "issues waiting for delete to complete: %v")
 		}
-		h.samplesResource = nil
-	} else {
-		srcfg = h.samplesResource
-	}
-	if srcfg != nil {
-		logrus.Println("SampleResource already received, not creating default")
-		return nil
 	}
-	// "4a" in the "startup" workflow, just create default
-	// resource and set up that way
-	srcfg = &v1alpha1.SamplesResource{}
-	srcfg.Name = v1alpha1.SamplesResourceName
-	srcfg.Kind = "SamplesResource"
-	srcfg.APIVersion = "samplesoperator.config.openshift.io/v1alpha1"
-	srcfg.Spec.Architectures = append(srcfg.Spec.Architectures, v1alpha1.X86Architecture)
-	srcfg.Spec.InstallType = v1alpha1.CentosSamplesDistribution
-	logrus.Println("creating default SamplesResource")
-	err = h.sdkwrapper.Create(srcfg)
-	if err != nil {
-		if !kerrors.IsAlreadyExists(err) {
-			return err
+
+	s, err := h.sdkwrapper.Get(v1alpha1.SamplesResourceName)
+	if s != nil && !kerrors.IsNotFound(err) {
+		logrus.Println("creating default SamplesResource")
+		err = h.sdkwrapper.Create(srcfg)
+		if err != nil {
+			if !kerrors.IsAlreadyExists(err) {
+				return nil, err
+			}
+			// in case there is some race condition
+			logrus.Println("got already exists error on create default")
 		}
-		logrus.Println("got already exists error on create default, just going to forgo this time vs. retry")
-		// just return the raw error and initiate the sdk's requeue/ratelimiter setup
+
 	}
 	h.deleteInProgress = false
-	return err
+	return srcfg, nil
 }
 
-func (h *Handler) manageDockerCfgSecret(deleted bool, samplesResource *v1alpha1.SamplesResource, secret *corev1.Secret) error {
-	if secret.Name != v1alpha1.SamplesRegistryCredentials {
+func (h *Handler) WaitingForCredential(srcfg *v1alpha1.SamplesResource) error {
+	if srcfg.ConditionTrue(v1alpha1.ImportCredentialsExist) {
 		return nil
 	}
 
-	if h.samplesResource == nil && h.waitingForCredential == nil {
-		if !deleted {
-			h.registrySecret = secret
+	// if trying to do rhel to the default registry.redhat.io registry requires the secret
+	// be in place since registry.redhat.io requires auth to pull; since it is not ready
+	// log error state
+	if srcfg.Spec.InstallType == v1alpha1.RHELSamplesDistribution &&
+		(srcfg.Spec.SamplesRegistry == "" || srcfg.Spec.SamplesRegistry == "registry.redhat.io") {
+		// currently do not attempt to avoid multiple settings of this conditions/message prior to an event
+		// where the situation is addressed
+		err := fmt.Errorf("Cannot create rhel imagestreams to registry.redhat.io without the credentials being available: %#v", srcfg)
+		h.processError(srcfg, v1alpha1.SamplesExist, corev1.ConditionFalse, err, "%v")
```

Comment:
this doesn't seem like the appropriate place to make this decision (that the samples do not exist).

this function's only purpose is to determine whether or not we are waiting for credentials, other areas of the code should decide whether the existence/non-existence of credentials has any implication regarding whether the samples exist.

Reply:
ah ... i mean to switch this to setting the import credentials to false, but then when we decided to set the conditions on the create default, I forgot to reconcile this.

addressed in next push

Reply:
sorry did add the import cred to false (with the detailed message) but just forgot to delete